js/privatebin.js

```diff
@@ -963,13 +962,17 @@ jQuery.PrivateBin = (function($) {
          * @returns {boolean}
          */
         function isStringContainsHtml(messageId) {
-            // An integer which specifies the type of the node. An Element node like <p> or <div>.
-            const elementNodeType = 1;
-
-            const div = document.createElement('div');
-            div.innerHTML = messageId;
-
-            return Array.from(div.childNodes).some(node => node.nodeType === elementNodeType);
+            // Use DOMParser to parse the string as HTML. DOMParser does not
+            // execute scripts nor load external resources when parsing, making
+            // it safer against XSS.
+            try {
+                const doc = new DOMParser().parseFromString(String(messageId), 'text/html');
+                return Array.from(doc.body.childNodes).some(node => node.nodeType === Node.ELEMENT_NODE);
+            } catch (e) {
+                // If parsing fails for any reason, consider it not HTML to avoid
+                // treating arbitrary strings as markup.
+                return false;
+            }
```

Comment:
Puh, this function keeps getting more and more complicated. Originally, before 0462da6, we simply checked for `args[0].indexOf('<a') !== -1` - no risk for any injection or execution of code and likely faster.

But we added `kbd` tags to the translations, so now there are message IDs with no anchors that still contain keyboard tags. How about we simplify this to:
```suggestion
            // message IDs are allowed to contain anchors, spans, keyboard and emphasis tags
            // we can recognize all of them by only checking for anchors and keyboard tags
            return args[0].indexOf('<a') !== -1 || args[0].indexOf('<kbd') !== -1;
```